app/src/main/java/com/guruthedev/instagram/ui/adapter/PostAdapter.kt

```diff
@@ -33,9 +33,9 @@ class PostAdapter(private val postList: ArrayList<Post>) :
     override fun onBindViewHolder(viewHolder: PostViewHolder, feedPost: Int) {
         val post = postList[feedPost]
         val binding = viewHolder.postBinding
-        binding.instagramPost= post
         binding.apply {
-        context?.let {
+            instagramPost = post
+            context?.let {
```

Comment:
null check context for context is not required here, as we are not passing context to addReadMore() function